packages/events/src/on.ts

```diff
@@ -2,14 +2,18 @@ import { resource, Operation } from 'effection';
 import { EventSource, addListener, removeListener } from './event-source';
 import { once } from './once';
 
+function isNotEmpty<T>(array: T[]): array is { shift(): T; } & Array<T> {
```

Comment:
I wonder if you could you type this as `array is [T,...T[]]`? as perhaps more directly typing it as "an array of at least one element" rather than an array whose shift method defiintely returns an item.

Also, can we move this helper function to the bottom of the file?

Reply:
I went ahead and updated it to use vanilla destructuring instead of `shift()` which seems to require no annotations at all.